### PR TITLE
implemeted a basic bot

### PR DESCRIPTION
**Pull request by fredmarques**
Still needs a database for session management

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+export BOT_TOKEN='209661714:AAEXCithVPoPQHksg9IKdLTjP3ty8qvUSdo'
```
Comment (fczuardi) [inline .env]: do not include tokens in the repo, thanks

File: setUp.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default {
```
Comment (fczuardi) [inline setUp.js]: please rename this file to setup.js (lowercase)

File: src/index.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+const app = new Telegraf(process.env.BOT_TOKEN, { username: botName });
 import debugMiddleware from './middlewares/debug';
 import photoUploadMiddleware from './middlewares/photoUpload';
+import tos from './middlewares/termsOfService';
 
-const app = new Telegraf(
-    process.env.BOT_TOKEN,
-    { username: process.env.BOT_USERNAME }
-);
+console.log('token', process.env.BOT_TOKEN);
```
Comment (fczuardi) [inline src/index.js]: this can be removed now

File: src/index.js [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 
 app.use(debugMiddleware);
 app.use(photoUploadMiddleware);
+app.use(tos);
 
-app.startPolling();
+app.startPolling(0);
```
Comment (fczuardi) [inline src/index.js:14]: what is the zero for?

File: src/middlewares/termsOfService.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import setUp from '../../setUp';
```
Comment (fczuardi) [inline src/middlewares/termsOfService.js]: rename to setup

File: src/middlewares/termsOfService.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+// TODO: change this to remote database
+const activeUsers = new Set();
+
+const tos = (ctx, next) => next().then(() => {
+    const { text, from } = ctx.update.message;
+
+    const isStart = text && text === '/start';
+    const isTosReply = text && (text === setUp.agreeText || text === setUp.declineText);
```
Comment (fczuardi) [inline src/middlewares/termsOfService.js]: same thing (setup)

File: src/middlewares/termsOfService.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    const hasAccepted = activeUsers.has(from.id);
+
+    if (isStart && !hasAccepted) {
+        ctx.reply(setUp.askForConfirmation, { reply_markup: {
```
Comment (fczuardi) [inline src/middlewares/termsOfService.js]: same thing (setup)

File: src/middlewares/termsOfService.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+            resize_keyboard: true,
+            one_time_keyboard: true,
+            keyboard: [
+                [{ text: setUp.agreeText }, { text: setUp.declineText }]]
```
Comment (fczuardi) [inline src/middlewares/termsOfService.js]: same thing (setup)

File: src/middlewares/termsOfService.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        }
+    });
+    } else if (isTosReply && !hasAccepted) {
+        if (text === setUp.agreeText) {
```
Comment (fczuardi) [inline src/middlewares/termsOfService.js]: same thing (setup)

File: src/middlewares/termsOfService.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+            activeUsers.add(from.id);
+            ctx.reply(setUp.notifyAgreed);
```
Comment (fczuardi) [inline src/middlewares/termsOfService.js]: same thing (setup)

File: src/middlewares/termsOfService.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        } else {
+            ctx.reply(setUp.notifyDeclined);
```
Comment (fczuardi) [inline src/middlewares/termsOfService.js]: same thing (setup)

**Review by fczuardi — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 4 above)
Comment (fredmarques) [reply src/index.js:14]: This is the amount of time between each poll, in seconds, as described [here](http://telegraf.js.org/telegraf.html#startpolling).

**Comment by fredmarques** [comment]
`setUp` becomes `setup` as suggested.

File: src/index.js (hunk 4 above)
Comment (fczuardi) [reply src/index.js:14]: pooling interval in seconds, zero is ok for now

**Review by fczuardi — APPROVED** [review]
👍

**Pull request merged**